src/ptr/shared.zig

```diff
@@ -232,6 +238,12 @@ pub fn WithOptions(comptime Pointer: type, comptime options: Options) type {
         fn getData(self: Self) if (info.isOptional()) ?*Data else *Data {
             return .fromValuePtr(self.#pointer);
         }
+
+        /// Assumes that the pointer is valid and was created by Shared.new().
```

Comment:
```suggestion
        /// Clones a shared pointer, given a raw pointer that originally came from a shared pointer.
        ///
        /// `pointer` must have come from a shared pointer (e.g., from `get` or `leak`), and the shared
        /// pointer from which it came must remain valid (i.e., not be deinitialized) at least until
        /// this function returns.
```

I think precise documentation is useful with unsafe functions